--- a/grammar/php.y
+++ b/grammar/php.y
@@ -158,3 +158,3 @@
     | T_BREAK | T_ARRAY | T_CALLABLE | T_EXTENDS | T_IMPLEMENTS | T_NAMESPACE | T_TRAIT | T_INTERFACE | T_CLASS
-    | T_CLASS_C | T_TRAIT_C | T_FUNC_C | T_METHOD_C | T_LINE | T_FILE | T_DIR | T_NS_C | T_FN
+    | T_CLASS_C | T_TRAIT_C | T_FUNC_C | T_METHOD_C | T_PROPERTY_C | T_LINE | T_FILE | T_DIR | T_NS_C | T_FN
     | T_MATCH | T_ENUM
@@ -310,5 +310,5 @@
       namespace_name
-          { $$ = Node\UseItem[$1, null, Stmt\Use_::TYPE_UNKNOWN]; $this->checkUseUse($$, #1); }
+          { $$ = Node\UseItem[$1, null, Stmt\Use_::TYPE_UNKNOWN]; }
     | namespace_name T_AS identifier_not_reserved
-          { $$ = Node\UseItem[$1, $3, Stmt\Use_::TYPE_UNKNOWN]; $this->checkUseUse($$, #3); }
+          { $$ = Node\UseItem[$1, $3, Stmt\Use_::TYPE_UNKNOWN]; }
 ;
@@ -317,5 +317,5 @@
       legacy_namespace_name
-          { $$ = Node\UseItem[$1, null, Stmt\Use_::TYPE_UNKNOWN]; $this->checkUseUse($$, #1); }
+          { $$ = Node\UseItem[$1, null, Stmt\Use_::TYPE_UNKNOWN]; }
     | legacy_namespace_name T_AS identifier_not_reserved
-          { $$ = Node\UseItem[$1, $3, Stmt\Use_::TYPE_UNKNOWN]; $this->checkUseUse($$, #3); }
+          { $$ = Node\UseItem[$1, $3, Stmt\Use_::TYPE_UNKNOWN]; }
 ;
@@ -1041,3 +1041,3 @@
     | T_CLONE clone_argument_list                           { $$ = Expr\FuncCall[new Node\Name($1, stackAttributes(#1)), $2]; }
-    | T_CLONE expr                                          { $$ = Expr\Clone_[$2]; }
+    | T_CLONE expr                                          { $$ = Expr\Clone_[$2]; $$->setAttribute('cloneExprLine', $this->tokens[$this->tokenPos]->line); }
     | variable T_PLUS_EQUAL expr                            { $$ = Expr\AssignOp\Plus      [$1, $3]; }
@@ -1068,3 +1068,3 @@
     | expr '^' expr                                         { $$ = Expr\BinaryOp\BitwiseXor[$1, $3]; }
-    | expr '.' expr                                         { $$ = Expr\BinaryOp\Concat    [$1, $3]; }
+    | expr '.' expr                                         { $$ = Expr\BinaryOp\Concat    [$1, $3]; $$->setAttribute('concatExprLine', $this->tokens[$this->tokenPos]->line); }
     | expr '+' expr                                         { $$ = Expr\BinaryOp\Plus      [$1, $3]; }
@@ -1102,5 +1102,8 @@
           }
+          if ($$ instanceof Expr\Ternary) {
+              $$->setAttribute('parenthesizedConditional', true);
+          }
       }
-    | expr '?' expr ':' expr                                { $$ = Expr\Ternary[$1, $3,   $5]; }
-    | expr '?' ':' expr                                     { $$ = Expr\Ternary[$1, null, $4]; }
+    | expr '?' expr ':' expr                                { $$ = Expr\Ternary[$1, $3,   $5]; $$->setAttribute('parenthesizedConditional', false); }
+    | expr '?' ':' expr                                     { $$ = Expr\Ternary[$1, null, $4]; $$->setAttribute('parenthesizedConditional', false); }
     | expr T_COALESCE expr                                  { $$ = Expr\BinaryOp\Coalesce[$1, $3]; }
@@ -1139,3 +1142,3 @@
     | T_PRINT expr                                          { $$ = Expr\Print_[$2]; }
-    | T_YIELD                                               { $$ = Expr\Yield_[null, null]; }
+    | T_YIELD                                               { $$ = Expr\Yield_[null, null]; $$->setAttribute('nullaryExprLine', $this->tokens[$this->tokenPos]->line); }
     | T_YIELD expr                                          { $$ = Expr\Yield_[$2, null]; }
@@ -1210,3 +1213,3 @@
     | name_readonly argument_list                           { $$ = Expr\FuncCall[$1, $2]; }
-    | callable_expr argument_list                           { $$ = Expr\FuncCall[$1, $2]; }
+    | callable_expr argument_list                           { $$ = Expr\FuncCall[$1, $2]; $$->setAttribute('callableExprLine', $this->tokens[$this->tokenStartStack[#2]]->line); }
     | class_name_or_var T_PAAMAYIM_NEKUDOTAYIM member_name argument_list
