--- a/grammar/php.y
+++ b/grammar/php.y
@@ -158,3 +158,3 @@
     | T_BREAK | T_ARRAY | T_CALLABLE | T_EXTENDS | T_IMPLEMENTS | T_NAMESPACE | T_TRAIT | T_INTERFACE | T_CLASS
-    | T_CLASS_C | T_TRAIT_C | T_FUNC_C | T_METHOD_C | T_LINE | T_FILE | T_DIR | T_NS_C | T_FN
+    | T_CLASS_C | T_TRAIT_C | T_FUNC_C | T_METHOD_C | T_PROPERTY_C | T_LINE | T_FILE | T_DIR | T_NS_C | T_FN
     | T_MATCH | T_ENUM
@@ -310,5 +310,5 @@
       namespace_name
-          { $$ = Node\UseItem[$1, null, Stmt\Use_::TYPE_UNKNOWN]; $this->checkUseUse($$, #1); }
+          { $$ = Node\UseItem[$1, null, Stmt\Use_::TYPE_UNKNOWN]; }
     | namespace_name T_AS identifier_not_reserved
-          { $$ = Node\UseItem[$1, $3, Stmt\Use_::TYPE_UNKNOWN]; $this->checkUseUse($$, #3); }
+          { $$ = Node\UseItem[$1, $3, Stmt\Use_::TYPE_UNKNOWN]; }
 ;
@@ -317,5 +317,5 @@
       legacy_namespace_name
-          { $$ = Node\UseItem[$1, null, Stmt\Use_::TYPE_UNKNOWN]; $this->checkUseUse($$, #1); }
+          { $$ = Node\UseItem[$1, null, Stmt\Use_::TYPE_UNKNOWN]; }
     | legacy_namespace_name T_AS identifier_not_reserved
-          { $$ = Node\UseItem[$1, $3, Stmt\Use_::TYPE_UNKNOWN]; $this->checkUseUse($$, #3); }
+          { $$ = Node\UseItem[$1, $3, Stmt\Use_::TYPE_UNKNOWN]; }
 ;
@@ -1102,5 +1102,8 @@
           }
+          if ($$ instanceof Expr\Ternary) {
+              $$->setAttribute('parenthesizedConditional', true);
+          }
       }
-    | expr '?' expr ':' expr                                { $$ = Expr\Ternary[$1, $3,   $5]; }
-    | expr '?' ':' expr                                     { $$ = Expr\Ternary[$1, null, $4]; }
+    | expr '?' expr ':' expr                                { $$ = Expr\Ternary[$1, $3,   $5]; $$->setAttribute('parenthesizedConditional', false); }
+    | expr '?' ':' expr                                     { $$ = Expr\Ternary[$1, null, $4]; $$->setAttribute('parenthesizedConditional', false); }
     | expr T_COALESCE expr                                  { $$ = Expr\BinaryOp\Coalesce[$1, $3]; }
